core/btdiscovery.cpp

```diff
@@ -17,6 +17,71 @@ namespace {
 }
 BTDiscovery *BTDiscovery::m_instance = NULL;
 
+struct modelPattern {
+	uint16_t    model;
+	const char *vendor;
+	const char *product;
+};
+static struct modelPattern model[] = {
+	{ 0x4552, "Oceanic", "Pro Plus X" },
+	{ 0x455A, "Aqualung", "i750TC" },
+	{ 0x4647, "Sherwood", "Sage" },
+	{ 0x4648, "Aqualung", "i300C" },
+	{ 0x4649, "Aqualung", "i200C" },
+	{ 0x4651, "Aqualung", "i770R" },
+	{ 0x4652, "Aqualung", "i550C" },
+	{ 0x4653, "Oceanic", "Geo 4.0" },
+	{ 0x4654, "Oceanic", "Veo 4.0" },
+	{ 0x4655, "Sherwood", "Wisdom 4" },
+	{ 0x4656, "Oceanic", "Pro Plus 4" },
+	{ 0x4743, "Aqualung", "i470TC" }
+};
+
+struct namePattern {
+	const char *prefix;
+	const char *vendor;
+	const char *product;
+};
+static struct namePattern name[] = {
+	// Shearwater dive computers
+	{ "Predator", "Shearwater", "Predator" },
+	// both the Petrel and Petrel 2 identify as "Petrel" as BT/BLE device
+	// but only the Petrel 2 is listed as available dive computer on iOS (which requires BLE support)
+	// so always pick the "Petrel 2" as product when seeing a Petrel
+	{ "Petrel", "Shearwater", "Petrel 2" },
+	{ "Perdix", "Shearwater", "Perdix" },
+	{ "Teric", "Shearwater", "Teric" },
+	{ "Peregrine", "Shearwater", "Peregrine" },
+	{ "NERD 2", "Shearwater", "NERD 2" },
+	{ "NERD", "Shearwater", "NERD" }, // order is important, test for the more specific one first
+	{ "Predator", "Shearwater", "Predator" },
+	{ "Predator", "Shearwater", "Predator" },
+	{ "Predator", "Shearwater", "Predator" },
```

Comment:
Just out of curiosity, why are there four identical Predators in this list?

Reply:
That's to make extra sure that... err... you know what... yeah... cut'n'paste error on my part.
Oops

Reply:
fixed